packages/components/src/modal/Modal.tsx

```diff
@@ -0,0 +1,183 @@
+import classNames from 'classnames';
+import React, {
+  ReactElement,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import ReactDOM from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+import ThemeExport from '../ThemeExport';
+
+interface ModalProps {
+  className?: string;
+  children?: ReactNode;
+  role?: string;
+  keyboard?: boolean;
+  isOpen?: boolean;
+  centered?: boolean;
+  size?: 'sm' | 'lg' | 'xl' | undefined;
+  onOpened?: () => void;
+  onClosed?: () => void;
+  toggle?: () => void;
+  'data-testid'?: string;
+}
+
+const Modal = ({
+  className = 'theme-bg-light',
+  children,
+  role = 'role',
+  keyboard = true,
+  isOpen = false,
+  centered = false,
+  size,
+  onOpened,
+  onClosed,
+  toggle,
+  'data-testid': dataTestId,
+}: ModalProps): ReactElement => {
+  const [isPortalOpen, setIsPortalOpen] = useState(false);
+
+  const element = useRef<HTMLElement>();
+
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent): void => {
+      switch (event.key) {
+        case 'Escape':
+          if (toggle) {
+            toggle();
+          }
+          break;
+        default:
+          break;
+      }
+    },
+    [toggle]
+  );
+
+  useEffect(
+    function addKeydownEventListener() {
+      if (isOpen && keyboard) {
+        window.addEventListener('keydown', handleKeyDown);
+        return () => window.removeEventListener('keydown', handleKeyDown);
+      }
+    },
+    [handleKeyDown, isOpen, keyboard]
+  );
+
+  useEffect(
+    function open() {
+      if (isOpen && onOpened) {
+        onOpened();
+      }
+    },
+    [onOpened, isOpen]
+  );
+
+  useEffect(
+    function closed() {
+      if (!isOpen && onClosed) {
+        onClosed();
+      }
+    },
+    [onClosed, isOpen]
+  );
+
+  useEffect(
+    function close() {
+      if (!isPortalOpen) {
+        if (element.current) {
+          document.body.removeChild(element.current);
+          element.current = undefined;
+        }
+      }
+    },
+    [isPortalOpen]
+  );
+
+  useEffect(
+    function open() {
+      if (isOpen && !element.current) {
+        element.current = document.createElement('div');
+        element.current.setAttribute(
+          'style',
+          'z-index: 1050; padding-right: 15px; display: block'
+        );
+        element.current.setAttribute('role', 'modal-container');
```

Comment:
This isn't a valid role - I don't think you need one? If you do, I would pick [presentation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role).